tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/fixer/FunctionNotFoundFixer.java

```diff
@@ -0,0 +1,48 @@
+package com.google.cloud.bigquery.utils.queryfixer.fixer;
+
+import com.google.cloud.bigquery.utils.queryfixer.entity.FixOption;
+import com.google.cloud.bigquery.utils.queryfixer.entity.FixResult;
+import com.google.cloud.bigquery.utils.queryfixer.entity.IToken;
+import com.google.cloud.bigquery.utils.queryfixer.errors.FunctionNotFoundError;
+import com.google.cloud.bigquery.utils.queryfixer.tokenizer.QueryTokenProcessor;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.util.Collections;
+
+/**
+ * The fixer class responsible for unrecognized functions. It fixes the error by (1) reading the
+ * similar function suggestion from the error message, and (2) replace the unrecognized function
+ * with the suggested one.
+ *
+ * <p>If the error message does not provide similar function suggestions, the fixer will directly
```

Comment:
Does this mean if the 'dry-run' error message doesn't provide similar function suggestions we will return error?

Reply:
For built-in function, yes. However, if the function is UDF, then logic will be implemented in future, like querying the server to find the UDF with similar names.

Reply:
OK. Please document the behavior somewhere so we know what to expect :).

Reply:
Yes, a TODO is put in the code. I will document this logic once it is implemented.